sway/desktop/render.c

```diff
@@ -1379,10 +1759,71 @@ static void render_seatops(struct sway_output *output,
 	}
 }
 
+static bool find_blurred_con_iterator(struct sway_container *con, void *data) {
+	struct sway_view *view = con->view;
+	if (!view || !con->blur_enabled) return false;
+	// Only test floating windows when xray is enabled
+	if (container_is_floating(con) && !config->blur_xray) return false;
+
+	struct wlr_surface *surface = view->surface;
+	if (surface->opaque) return false;
+
+	pixman_box32_t surface_box = {
+		.x1 = 0,
+		.y1 = 0,
+		.x2 = surface->current.width,
+		.y2 = surface->current.height,
+	};
+
+	pixman_region32_t inverse_opaque;
+	pixman_region32_init(&inverse_opaque);
+	// Gets the non opaque region
+	pixman_region32_copy(&inverse_opaque, &surface->opaque_region);
+	pixman_region32_inverse(&inverse_opaque, &inverse_opaque, &surface_box);
+	pixman_region32_intersect_rect(&inverse_opaque, &inverse_opaque, 0, 0,
+			surface->current.width,
+			surface->current.height);
+	bool not_empty = pixman_region32_not_empty(&inverse_opaque);
+	pixman_region32_fini(&inverse_opaque);
+	return not_empty;
```

Comment:
Q: if the surface is not opaque, shouldn't we just return `true`? Is the extra checking for non opaque region necessary?

Reply:
Good catch! Then we should just be able to provide the function with the translucent region instead of the opaque region

Reply:
Provide which function the translucent region?

Reply:
The render_blur function (not sure of the name but the function which this code lives in)

Reply:
I think we may be talking about different parts of the code, but if providing the translucent region over the opaque region elsewhere helps, that's great too!